waspc/data/Generator/templates/react-app/src/actions/index.js

```diff
@@ -0,0 +1,74 @@
+import {
+  useMutation,
+  useQueryClient,
+} from 'react-query'
+
+export { configureQueryClient } from '../queryClient'
+
+export function useAction(actionFn, actionOptions) {
+  const queryClient = useQueryClient();
+
+  // TODO(filip): Do we allow all react query options or should we consider a list of allowed options?
+  let options = {}
+
+  if (actionOptions?.optimisticUpdates) {
+    const optimisticUpdatesConfig = makeOptimisticUpdatesConfig(actionOptions.optimisticUpdates)
+    options = makeOptimisticUpdateOptions(queryClient, optimisticUpdatesConfig)
+  }
+
+  return useMutation(actionFn, options)
+}
+
+
+function makeOptimisticUpdatesConfig(optimisticUpdatesConfig) {
+  return optimisticUpdatesConfig.map(({ getQuery, ...rest }) => ({
+    getQuery: (item) => parseQueryKey(getQuery(item)),
+    ...rest,
+  }))
+}
+
+function makeOptimisticUpdateOptions(queryClient, optimisticUpdatesConfig) {
+  async function onMutate(item) {
+    const queriesToUpdate = optimisticUpdatesConfig.map(({ getQuery, ...rest }) => ({
+      queryKey: getQuery(item),
+      ...rest,
+    }))
```

Comment:
You say queriesToUpdate, but you return { queryKey, ... rest }. That doesn't look like query? I think I am missing a ton of context here, but maybe you could alleviate that with some comments?

Reply:
Yes, I got a bit liberal with the word `query` here. The object contains everything we need to perform an optimistic update. I'll do an additional pass that takes care of the naming and code style before requesting a final review.